### DISCUSSION TITLE
Multiple Components on Page Calls Eloquent User Provider with DB Hits for Every Component Loaded

### DISCUSSION THREAD

**Discussion by awizemann** [livewire/livewire#2193] [4. General]
I am unsure if this is the right place for this, but I noticed a performance issue for larger-scale SPA's that use livewire. I have an app built with 5 main components that can load sections from each component into the main view using listeners and events. When I initially load the page, the Auth libraries do their thing and query the DB to get what they need. One a subsequent call through an Event/Listener relationship, the same calls happen for every single component call, resulting in 5 similar requests (more if I call the components with what I am loading). Here is the backtrace I see duplicated often:

![Screen Shot 2020-12-09 at 10 37 57 AM](https://user-images.githubusercontent.com/319078/101672383-a824cc80-3a0a-11eb-9aac-df1bcd265f7a.png)

I am unsure if this is fixable based on the architecture of how Laravel works with Auth, but any help would be appreciated if something can be done to reduce the DB traffic.

**Comment by joshhanley** [comment]
@awizemann any change you could share a sample repo so we can run some tests to see if it's a Livewire thing or whether it's Laravel?

**Reply by awizemann** [reply]
I can't share broadly because it's a commercial application that's in development. It looks to be the Auth system of Laravel.
